File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/PriceInfo.java
package za.co.woolworths.financial.services.android.models.dto;

import java.io.Serializable;

/**
 * Created by W7099877 on 2018/02/08.
 */

public class PriceInfo {

	public double amount;
	public double listPrice;
	public double rawTotalPrice;
	public double salePrice;
	public boolean onSale;
	public double totalDiscount;

	public double getAmount() {
		return amount;
	}

	public void setAmount(double amount) {
		this.amount = amount;
	}

	public double getListPrice() {
		return listPrice;
	}

	public void setListPrice(double listPrice) {
		this.listPrice = listPrice;
	}

	public double getRawTotalPrice() {
		return rawTotalPrice;
	}

	public void setRawTotalPrice(double rawTotalPrice) {
		this.rawTotalPrice = rawTotalPrice;
	}

	public double getSalePrice() {
		return salePrice;
	}

	public void setSalePrice(double salePrice) {
		this.salePrice = salePrice;
	}

	public boolean isOnSale() {
		return onSale;
	}

	public void setOnSale(boolean onSale) {
		this.onSale = onSale;
	}

	public double getDiscountedAmount() {
		if (totalDiscount == 0 && amount != rawTotalPrice) {
			totalDiscount = (rawTotalPrice - amount);
		}
		return totalDiscount;
	}

	public void setDiscountedAmount(double discountedAmount) {
		this.totalDiscount = discountedAmount;
	}
}

File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/cart/PriceInfo.kt
package za.co.woolworths.financial.services.android.models.dto.cart

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue

@Parcelize
data class PriceInfo(
    var discounted: Boolean? = null,
    var amount: Double = 0.00,
    var rawTotalPrice: @RawValue Any = 0.00,
    var simplePromo: @RawValue Any = 0,
    var salePrice: @RawValue Any = 0,
    var itemSavings: @RawValue Any = 0.00,
    var wrewardsDiscount: @RawValue Any = 0,
    var staffDiscount: @RawValue Any = 0,
    var voucherDiscount: @RawValue Any = 0,
    var promoPrice: @RawValue Any = 0.00,
    var onSale: Boolean? = null,
    var totalDiscount: @RawValue Any = 0,
    var otherDiscount: @RawValue Any = 0,
    var listPrice: @RawValue Any = 0.00,
    var simplePromotion: Boolean? = null,
    var promoCodeDiscount: @RawValue Any = 0
) : Parcelable